tools/IoTEdgeDevOps/DevOpsLib/AzureActiveDirectory.cs

```diff
@@ -11,83 +11,86 @@ namespace DevOpsLib
+                var client = new HttpClient();
+                client.BaseAddress = new Uri(requestUri);
 
-                    var requestBody = new List<KeyValuePair<string, string>>();
-                    requestBody.Add(new KeyValuePair<string, string>("client_id", azureActiveDirClientId));
-                    requestBody.Add(new KeyValuePair<string, string>("client_secret", azureActiveDirClientSecret));
-                    requestBody.Add(new KeyValuePair<string, string>("grant_type", grantType));
-                    requestBody.Add(new KeyValuePair<string, string>("resource", azureResource));
+                var requestBody = new List<KeyValuePair<string, string>>();
+                requestBody.Add(new KeyValuePair<string, string>("client_id", azureActiveDirClientId));
+                requestBody.Add(new KeyValuePair<string, string>("client_secret", azureActiveDirClientSecret));
+                requestBody.Add(new KeyValuePair<string, string>("grant_type", grantType));
+                requestBody.Add(new KeyValuePair<string, string>("resource", azureResource));
 
-                    HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, "");
-                    // By default, if FormUrlEncodedContent() is used, the "Content-Type" is set to "application/x-www-form-urlencoded"
-                    // which can be explicitly written : request.Content.Headers = new MediaTypeHeaderValue("application/x-www-form-urlencoded");
-                    request.Content = new FormUrlEncodedContent(requestBody);
+                HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, "");
+                // By default, if FormUrlEncodedContent() is used, the "Content-Type" is set to "application/x-www-form-urlencoded"
+                // which can be explicitly written : request.Content.Headers = new MediaTypeHeaderValue("application/x-www-form-urlencoded");
+                request.Content = new FormUrlEncodedContent(requestBody);
 
-                    HttpResponseMessage response = await client.SendAsync(request).ConfigureAwait(false);
-                    response.EnsureSuccessStatusCode();
-                    string responseMsg = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                HttpResponseMessage response = await client.SendAsync(request).ConfigureAwait(false);
+                response.EnsureSuccessStatusCode();
+                string responseMsg = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
 
-                    lock (locker)
-                    {
-                        var responseJson = JObject.Parse(responseMsg);
-                        this.azureResource = azureResource;
-                        this.accessTokenExpiration = DateTime.UtcNow.AddSeconds((double)responseJson["expires_on"] - 1);
-                        this.accessToken = (string)responseJson["access_token"];
-                    }
+                lock (locker)
```

Comment:
Need to move this lock statement above if block (line 47); and you need to Semaphore. https://blog.cdemi.io/async-waiting-inside-c-sharp-locks/